libs/wingc/src/parser.rs

```diff
@@ -154,6 +154,13 @@ impl Parser<'_> {
 	fn build_scope<'a>(&self, scope_node: &Node) -> Scope {
 		let mut cursor = scope_node.walk();
 
+		// check if the last child is a missing closing brace
+		if let Some(last_child) = scope_node.children(&mut cursor).last() {
+			if last_child.kind() == "}" && last_child.is_missing() {
+				_ = self.add_error::<Node>(format!("'}}' expected."), &last_child);
```

Comment:
Aside - does anyone know why some methods in the parser module return an `AST type` (like this method) while other methods return a `DiagnosticResult<AST type>`? is one of them more correct?

Reply:
Generally `DiagnosticResult` is more correct. If using the unwrapped one, it's because either it can't fail or we don't handle the failure. The latter is more likely 🥲 

In the case of this method, building a scope can't really _fail_. It can be empty, but if you already invoked this method you're going to get some kind of scope back.